sec_certs/sample/common_criteria.py

```diff
@@ -770,6 +762,8 @@ def _fix_ocsi_cert_id(cert_id: str) -> str:
         return new_cert_id
 
     def get_cert_laboratory(self) -> str:
+        if not self.heuristics.cert_id:
+            raise ValueError("Cert ID was None but cert laboratory was to be computed based on its value.")
```

Comment:
@GeorgeFI I needed to add these lines due to MyPy (it was not properly checking `heuristics` before, it does now). Is this ok with you?

Also, is there any specific reason why `heuristics.directly_affected_by` is a `list`, while other dependency attributes are `set`s? If possible, we would like to stick to sets to avoid unnecessary difs.